src/integer/z/arithmetic/add.rs

```diff
@@ -46,8 +47,101 @@ impl Add for &Z {
     }
 }
 
-arithmetic_trait_borrowed_to_owned!(Add, add, Z);
-arithmetic_trait_mixed_borrowed_owned!(Add, add, Z);
+arithmetic_trait_borrowed_to_owned!(Add, add, Z, Z, Z);
+arithmetic_trait_mixed_borrowed_owned!(Add, add, Z, Z, Z);
+arithmetic_between_types!(Add, add, Z, i64 i32 i16 i8 u64 u32 u16 u8);
+
+#[cfg(test)]
+mod test_add_between_types {
+
+    use crate::integer::Z;
+    use std::str::FromStr;
+
+    /// testing addition between different types
+    #[test]
+    fn add() {
+        let a: Z = Z::from_str("42").unwrap();
```

Comment:
Please also do this for the others, just my personal preference
```suggestion
        let a: Z = Z::from("42");
```